File: src/main/java/com/test/taskmanager/app/view/ProcessPriorityScreen.java
package com.test.taskmanager.app.view;

import static com.test.taskmanager.domain.Priority.HIGH;
import static com.test.taskmanager.domain.Priority.LOW;
import static com.test.taskmanager.domain.Priority.MEDIUM;

import com.test.taskmanager.domain.Priority;
import de.codeshelf.consoleui.prompt.ListResult;
import de.codeshelf.consoleui.prompt.PromtResultItemIF;
import io.vavr.control.Try;
import java.util.Map;
import lombok.Value;
import lombok.experimental.SuperBuilder;

@Value
@SuperBuilder
public class ProcessPriorityScreen extends AbstractScreen<Priority> {

    private static final String LOW_ACTION = "low";
    private static final String MEDIUM_ACTION = "medium";
    private static final String HIGH_ACTION = "high";

    @Override
    public Priority invoke() {
        promptBuilder.createListPrompt()
            .name(ACTION)
            .message("Add process priority:")
            .newItem(LOW_ACTION).text("1 - Low").add()
            .newItem(MEDIUM_ACTION).text("2 - Medium").add()
            .newItem(HIGH_ACTION).text("3 - High").add()
            .newItem(BACK_ACTION).text("4 - Back").add()
            .addPrompt();

        Map<String, ? extends PromtResultItemIF> result = Try.of(() -> prompt.prompt(promptBuilder.build()))
            .get();

        ListResult action = (ListResult) result.get(ACTION);

        switch (action.getSelectedId()) {
            case LOW_ACTION:
                return LOW;
            case MEDIUM_ACTION:
                return MEDIUM;
            case HIGH_ACTION:
                return HIGH;
            case BACK_ACTION:
                AddProcessScreen.builder()
                    .taskManager(taskManager)
                    .build()
                    .invoke();
        }

        return LOW;
    }
}

File: src/main/kotlin/com/test/taskmanager/app/view/ProcessPriorityScreen.kt
package com.test.taskmanager.app.view

import com.test.taskmanager.domain.Priority
import com.test.taskmanager.domain.Priority.*
import com.test.taskmanager.service.TaskManager
import de.codeshelf.consoleui.prompt.ListResult
import de.codeshelf.consoleui.prompt.PromtResultItemIF
import io.vavr.control.Try

class ProcessPriorityScreen(taskManager: TaskManager) : AbstractScreen<Priority>(taskManager) {
    override fun invoke(): Priority {
        promptBuilder.createListPrompt()
                .name(ACTION)
                .message("Add process priority:")
                .newItem(LOW_ACTION).text("1 - Low").add()
                .newItem(MEDIUM_ACTION).text("2 - Medium").add()
                .newItem(HIGH_ACTION).text("3 - High").add()
                .newItem(BACK_ACTION).text("4 - Back").add()
                .addPrompt()

        val result: Map<String, PromtResultItemIF> = Try.of { prompt.prompt(promptBuilder.build()) }
                .get()
        val action = result[ACTION] as ListResult

        if (BACK_ACTION == action.selectedId) {
            AddProcessScreen(taskManager).invoke()
        }

        return when (action.selectedId) {
            LOW_ACTION -> LOW
            MEDIUM_ACTION -> MEDIUM
            HIGH_ACTION -> HIGH
            else -> LOW
        }
    }

    companion object {
        private const val LOW_ACTION = "low"
        private const val MEDIUM_ACTION = "medium"
        private const val HIGH_ACTION = "high"
    }
}